test/Dialect/FIRRTL/infer-widths-errors.mlir

```diff
@@ -121,3 +121,16 @@ firrtl.circuit "Foo"  {
     firrtl.connect %w, %0 : !firrtl.bundle<x: uint>, !firrtl.bundle<x: uint>
   }
 }
+
+// -----
+// Unsatisfiable widths through AttachOp on analog types should error.
+// https://github.com/llvm/circt/issues/4786
+firrtl.circuit "AnalogWidths" {
+  // expected-error @+1 {{uninferred width: port "a" cannot satisfy all width requirements}}
+  firrtl.module @AnalogWidths(in %a: !firrtl.analog, out %b: !firrtl.analog<2>, out %c: !firrtl.analog<1>) {
+    firrtl.attach %a, %b : !firrtl.analog, !firrtl.analog<2>
+    // expected-note @+2 {{width is constrained to be at least 2 here:}}
```

Comment:
Great diagnostics, thanks!! :rocket: 

FWIW can use `@below` for both of these, helpful to avoid having to count or just to not think about it :wink:.

Reply:
Sweet, did not know about `@below`! Thanks!